caffe2/core/tensor.h

```diff
@@ -102,6 +102,25 @@ class CAFFE2_API Tensor final {
     return x;
   }
 
+  /**
+   * Clone self as a Tensor that share the same Storage
+   */
+  Tensor Alias() const {
+    Tensor x(sizes(), GetDevice());
+    if (!dtype_initialized()) {
+      C10_LOG_EVERY_MS(WARNING, 1000) <<
+                   "Cloning a tensor that don't have a data type (did you call mutable_data<T> on the tensor?)";
+    }
+    AT_ASSERTM(
+        storage_initialized(),
+        "Cloning a tensor that has no content and has size > 0");
+    x.impl_->set_storage(storage());
+    x.raw_mutable_data(dtype());
```

Comment:
Does this do the right thing when dtype is not initialized? I'm not convinced...

Reply:
Looking again, we probably don't need to call raw_mutable_data, since set_storage already set data_type_ of TensorImpl to be that of Storage.